src/Plugin.php

```diff
@@ -27,7 +27,7 @@ class Plugin {
    * Plugin init.
    */
   public static function init() {
-    if (is_admin()) {
+    if (is_admin() || strpos($_SERVER['REQUEST_URI'],'.xml') !== FALSE) {
```

Comment:
This is not the correct way to check whether the current request produces a feed.

Reply:
Could you give me a little hint on that ?

Reply:
It's most often sufficient to take the topic…

> …check whether the current request produces a feed.

…and prefix it with "wordpress":

https://www.google.de/search?q=wordpress+check+whether+the+current+request+produces+a+feed

So your condition turns into:

```diff
-    if (is_admin() || strpos($_SERVER['REQUEST_URI'],'.xml') !== FALSE) {
+    if (is_admin() || !is_feed()) {
```

Reply:
....  simple like that (!). Forgot about that function. OK - will change it! Thank you!

Reply:
I tested on local with this solution, and with !is_feed() solution, plugin don't display message on other queries and subpages.  I will try to look into that direction now.

Reply:
It's because the condition is wrong. It should be

Reply:
1. I guess Sunny wrongly wrote it. It should be `is_feed()` and not `!is_feed()` because in case of XML feeds this will return `TRUE` so will run the early return. Am I right?
2. Also, using `is_feed()` here does not help us. Could be too late?

Reply:
yes, i've do as well with feed, but then message is display message in sitemap anyway - so or i can dig into is_feed or we could stay with previous version ?

Reply:
After digging deeper i found different solution: 

Instead of 
> add_action('init', __NAMESPACE__ . '\Plugin::init');
(this action always returns false value for condition like is_page(), is_feed() etc. )
i gave 
> add_action('wp', __NAMESPACE__ . '\Plugin::init');
which is recommended here: https://codex.wordpress.org/Plugin_API/Action_Reference

And then condifition i've mentioned before are working properly. But also pros of that solution: it's not adding mentioned message to sitemap. Just in case i've still added condition
> is_admin() || is_feed()
 if plugin will be used on different websites. 

So i think it fixes our problem in correct way i guess...  ?

fixup on its way...!